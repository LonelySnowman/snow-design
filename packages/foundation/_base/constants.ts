export const BASE_CLASS_PREFIX = 'snow';
