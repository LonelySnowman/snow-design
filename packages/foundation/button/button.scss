@import './variables';

$module: #{$prefix}-button;

.#{$module} {
  height: 32px;
  padding: $button-padding;
  cursor: pointer;
  border: 0 transparent solid;
  border-radius: 6px;
  outline: none;
  transition: all linear 0.1s;

  &-default {
    color: $button-color_default-text-default;
    background-color: $button-color_default-bg-default;

    &:hover {
      background-color: $button-color_default-bg-hover;
    }

    &:active {
      background-color: $button-color_default-bg-active;
    }
  }

  &-primary {
    color: $button-color_primary-text-default;
    background-color: $button-color_primary-bg-default;

    &:hover {
      background-color: $button-color_primary-bg-hover;
    }

    &:active {
      background-color: $button-color_primary-bg-active;
    }
  }

  &-warning {
    color: $button-color_warning-text-default;
    background-color: $button-color_warning-bg-default;

    &:hover {
      background-color: $button-color_warning-bg-hover;
    }

    &:active {
      background-color: $button-color_warning-bg-active;
    }
  }

  &-danger {
    color: $button-color_danger-text-default;
    background-color: $button-color_danger-bg-default;

    &:hover {
      background-color: $button-color_danger-bg-hover;
    }

    &:active {
      background-color: $button-color_danger-bg-active;
    }
  }
}
