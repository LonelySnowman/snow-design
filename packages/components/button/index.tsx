import React, { ReactNode } from 'react';
import '@snow-design/foundation/button/button.scss';
import { cssClasses } from '@snow-design/foundation/button/constants';
import classNames from 'classnames';
import { CssProps } from '@snow-design/components/_types';

const prefixCls = cssClasses.PREFIX;

export type ThemeType = 'default' | 'primary' | 'warning' | 'danger';

export interface ButtonProps extends Omit<React.ButtonHTMLAttributes<HTMLElement>, 'type'>, CssProps {
    children?: ReactNode;
    type?: ThemeType;
}

const Button: React.FC<ButtonProps> = (props) => {
    const { children, style, className, type = 'default', ...baseProps } = props;

    const classes = classNames(className, prefixCls, {
        [`${prefixCls}-${type}`]: type,
    });

    return (
        <button {...baseProps} style={style} className={classes}>
            <span>{children}</span>
        </button>
    );
};

export default Button;
