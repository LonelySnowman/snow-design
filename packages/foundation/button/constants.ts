import { BASE_CLASS_PREFIX } from '../_base/constants';

export const cssClasses = {
    PREFIX: `${BASE_CLASS_PREFIX}-button`,
};
