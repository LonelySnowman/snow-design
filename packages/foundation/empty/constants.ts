import { BASE_CLASS_PREFIX } from '../_base/constants';

const cssClasses = {
    PREFIX: `${BASE_CLASS_PREFIX}-empty`
};

export {
    cssClasses,
};
