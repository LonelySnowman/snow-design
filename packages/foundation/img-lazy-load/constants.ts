import { BASE_CLASS_PREFIX } from '../_base/constants';

const cssClasses = {
    PREFIX: `${BASE_CLASS_PREFIX}-img-lazy-load`,
};

export { cssClasses };
