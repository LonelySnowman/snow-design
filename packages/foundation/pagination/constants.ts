import { BASE_CLASS_PREFIX } from '../_base/constants';

const cssClasses = {
    PREFIX: `${BASE_CLASS_PREFIX}-pagination`
};

const strings = {};

const numbers = {
    PAGE_SHOW_MAX: 7,
    REST_PAGE_SHOW_MAX: 5,
    DEFAULT_PAGE_SIZE: 10,
    PAGE_SIZE_OPTION: [10, 20, 40, 100],
    REST_PAGE_MAX_SIZE: 1000000,
}

export { cssClasses, strings, numbers };
