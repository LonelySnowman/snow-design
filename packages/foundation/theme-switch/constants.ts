import { BASE_CLASS_PREFIX } from '../_base/constants';

export type ThemeType = 'dark' | 'light';

const cssClasses = {
    PREFIX: `${BASE_CLASS_PREFIX}-theme-switch`,
};

export { cssClasses };
