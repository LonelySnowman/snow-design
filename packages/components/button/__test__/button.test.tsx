import { render } from '@testing-library/react';
import '@testing-library/jest-dom'
import React from 'react';
import Button from '../index'
import { cssClasses } from '@snow-design/foundation/button/constants';

describe('Button', () => {
    const prefix = cssClasses.PREFIX;
    it('Button rendered', () => {
        const { container } = render(<Button style={{ color: 'red' }}>Test</Button>);
        expect(container.querySelector<HTMLDivElement>(`.${prefix}`)?.style.color).toBe('red');
    });
    it('Type have class', () => {
        const { container } = render(<Button type="primary">Test</Button>);
        expect(container.firstChild).toHaveClass(`${prefix}-primary`)
    });
});
