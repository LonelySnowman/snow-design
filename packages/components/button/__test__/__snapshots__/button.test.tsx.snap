// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  class="snow-button snow-button-default"
>
  <span>
    SnowDesign
  </span>
</button>
`;
